components/infobox/extensions/commons/infobox_extensions_prize_pool.lua

```diff
@@ -0,0 +1,161 @@
+local LANG = mw.language.new('en')
+local CATEGRORY = '[[Category:Tournaments with invalid prize pool]]'
+
+function PrizePoolCurrency.display(args)
+	local currency = string.upper(args.currency or USD)
+	local date = PrizePoolCurrency._cleanDate(args.date or NOW)
+	local text = args.text or 'Currency exchange rate taken from exchangerate.host'
+	local prizepool = PrizePoolCurrency._cleanValue(args.prizepool)
+	local prizepoolUsd = PrizePoolCurrency._cleanValue(args.prizepoolusd)
+	local currencyRate = tonumber(args.rate)
+	local setVariables = Logic.emptyOr(args.setvariables, true)
+
+	if not prizepool and not prizepoolUsd then
+		if Namespace.isMain() then
+			return (args.prizepool or args.prizepoolUsd or '')
+				.. '[[Category:Tournaments with invalid prize pool]]'
+		else
+			return (args.prizepool or args.prizepoolUsd or '')
+		end
+	end
+
+	if not date then
+		date = NOW
+		mw.log('Infobox league: Invalid currency exchange date -> default date (' .. date .. ')')
+	elseif date > NOW then
+		date = NOW
+	end
+
+	if currency == USD and String.isEmpty(prizepoolUsd) then
+		return PrizePoolCurrency._errorMessage('Need valid currency')
+	elseif currency == USD then
+		prizepoolUsd = LANG:formatNum(math.floor(prizepoolUsd))
+	else
+		local errorMessage
+		errorMessage, prizepool, prizepoolUsd, currencyRate = PrizePoolCurrency._exchange{
+			currency = currency,
+			currencyRate = currencyRate,
+			date = date,
+			prizepool = prizepool,
+			prizepoolUsd = prizepoolUsd,
+		}
+		if errorMessage then
+			return PrizePoolCurrency._errorMessage(errorMessage)
+		end
+	end
+
+	if setVariables then
+		if String.isNotEmpty(args.currency) and currencyRate then
+			Variables.varDefine(currency .. '_rate', currencyRate)
+		end
+		if String.isNotEmpty(args.currency) then
+			Variables.varDefine('tournament_currency', currency:upper())
+		end
+
+		Variables.varDefine('tournament_currency_date', date)
+		Variables.varDefine('tournament_currency_text', text)
+		Variables.varDefine('tournament_prizepoollocal', prizepool)
+
+		local prizepoolUsdValue = string.gsub(prizepoolUsd or '', ',', '')
+		Variables.varDefine('tournament_prizepoolusd', prizepoolUsdValue)
+
+		-- legacy compatibility
+		Variables.varDefine('tournament_currency_rate', currencyRate or '')
+		Variables.varDefine('tournament_prizepool_local', prizepool)
+		Variables.varDefine('tournament_prizepool_usd', prizepoolUsd)
+		Variables.varDefine('currency', args.currency and currency)
+		Variables.varDefine('currency date', date)
+		Variables.varDefine('currency rate', currencyRate)
+		Variables.varDefine('prizepool', prizepool)
+		Variables.varDefine('prizepool usd', prizepoolUsd)
+		Variables.varDefine('tournament_prizepool', prizepoolUsdValue)
+	end
+
+	local display = Currency.display(USD, prizepoolUsd, {formatValue = false, setVariables = false})
+	if String.isNotEmpty(prizepool) then
+		display = Currency.display(currency, prizepool, {formatValue = false, setVariables = true})
```

Comment:
Not formatted values?

Reply:
the `formatValue` in module currency rounds by precision 2 (after the digit)
this module rounds to full $/€/...
but we can remove the `formatValue = false` stuff (irrelvant if nil or false)

Reply:
Sounds good